### PR TITLE
Remove InPullConnector class

### LINKED ISSUE #2: Remove InPullConnector

### PR DESCRIPTION
**Pull request by scymtym**
This supersedes and extends #5.

Will fix #2.

### CODE REVIEW

**Comment by scymtym** [comment]
Now with mock-based test for connector deactivation. Should be ready.

File: test/rsb/transport/ConnectorTest.cpp [hunk 1]
```diff
@@ -66,15 +66,62 @@ int pullInConnectorTest() {
...
+class MockBus : public rsb::transport::spread::Bus {
+public:
+    MockBus() {};
+
+    MOCK_CONST_METHOD0(getTransportURL, const std::string());
+
+    MOCK_METHOD0(activate, void());
+    MOCK_METHOD0(deactivate, void());
+
+    MOCK_METHOD2(addSink, void(const rsb::Scope& scope, rsb::transport::spread::Bus::SinkPtr sink));
```
Comment (languitar) [inline test/rsb/transport/ConnectorTest.cpp]: Didn't we have something like an 80 characters line length restriction usually?

File: test/rsb/transport/ConnectorTest.h [hunk 2]
```diff
@@ -41,13 +42,19 @@
...
+            boost::function<rsb::transport::InConnectorPtr(rsb::transport::spread::BusPtr)> createInConnectorWithBus,
+            boost::function<rsb::transport::OutConnectorPtr(rsb::transport::spread::BusPtr)> createOutConnectorWithBus) :
+        createConnectingInConnector(createConnectingInConnector),
+        createConnectingOutConnector(createConnectingOutConnector),
+        createInConnectorWithBus(createInConnectorWithBus),
+        createOutConnectorWithBus(createOutConnectorWithBus){
     }
-    boost::function<rsb::transport::InConnectorPtr()> createInConnector;
-    boost::function<rsb::transport::OutConnectorPtr()> createOutConnector;
+    boost::function<rsb::transport::InConnectorPtr()> createConnectingInConnector;
```
Comment (languitar) [inline test/rsb/transport/ConnectorTest.h]: Maybe some comments here to describe what the new functions are and how they differ from the renamed existing ones?

**Review by languitar — CHANGES_REQUESTED** [review]
Just a few minor things.

File: test/rsb/transport/ConnectorTest.cpp (hunk 1 above)
Comment (scymtym) [reply test/rsb/transport/ConnectorTest.cpp]: Mostly addressed.

File: test/rsb/transport/ConnectorTest.h (hunk 2 above)
Comment (scymtym) [reply test/rsb/transport/ConnectorTest.h]: Done.

**Review by languitar — APPROVED** [review]

**Pull request merged**